OptimizelySDKCore/OptimizelySDKCore/OPTLYDecisionService.m

```diff
@@ -163,8 +164,9 @@ - (NSString *)getBucketingId:(NSString *)userId
     NSString *bucketingId = userId;
     // If the bucketing ID key is defined in attributes, then use that
     // in place of the userID for the murmur hash key
-    if (![OPTLYDecisionService isEmptyString:attributes[OptimizelyBucketId]]) {
-        bucketingId = (NSString *)attributes[OptimizelyBucketId];
+    
+    for (NSString *validBucketingId = [attributes[OptimizelyBucketId] getValidString]; validBucketingId != nil; validBucketingId = nil) {
```

Comment:
@yasirfolio3 Why for loop is used instead of if.

Reply:
in case of if getValidString != nil, the getValidString method would be called twice, the above implemented way is just an alternate way of using "if let" from swift where getValidString would only be required to call once, attached is the reference:

http://blog.cristik.com/2016/01/if-let-statement-in-objective-c/